staging/src/kubevirt.io/client-go/kubecli/handler.go

```diff
@@ -75,9 +74,10 @@ type virtHandler struct {
 }
 
 type virtHandlerConn struct {
-	pod  *v1.Pod
-	err  error
-	port int
+	pod        *v1.Pod
+	err        error
+	port       int
+	httpClient *http.Client
```

Comment:
> Do you have any idea why the connections aren't being re-used? Am I missing something here?

Seems that `virtHandlerConn` is re-instantiated each time a new request is processed:

https://github.com/kubevirt/kubevirt/blob/a497ef810ea58b4a7ad25089c6a37bc3bd7786db/staging/src/kubevirt.io/client-go/kubecli/handler.go#L92-L93

https://github.com/kubevirt/kubevirt/blob/a497ef810ea58b4a7ad25089c6a37bc3bd7786db/pkg/virt-api/rest/subresource.go#L193-L198

That way, the client is not reused. Probably we need to keep `virtHandlerConn` objects somewhere.

Reply:
UPD: I tested locally with a singleton client. I observe only one connection is created. Seems that setting `MaxConnsPerHost` is not necessary.

Reply:
Once again thanks a lot @vasiliy-ul for diving into the problem and providing great insights!

I've tried to cache virt-handler connections as well in https://github.com/kubevirt/kubevirt/pull/8948/commits/3104bb01af4cc21fae11b3929a3d777352aa0a05. However, I still don't see that the connections are being re-used:
```bash
bash-5.1# while true; do CONS=`ss -ntlap | grep 8186 | wc -l`; echo "Number of connections: $CONS"; sleep 2; done
Number of connections: 1
Number of connections: 2
Number of connections: 7
Number of connections: 13
Number of connections: 18
Number of connections: 23
Number of connections: 32
Number of connections: 36
Number of connections: 41
Number of connections: 47
Number of connections: 51
Number of connections: 53
# ... Stays 53 for about 45 seconds
Number of connections: 53
Number of connections: 47
Number of connections: 41
Number of connections: 37
Number of connections: 31
Number of connections: 31
Number of connections: 22
Number of connections: 18
Number of connections: 13
Number of connections: 8
Number of connections: 4
Number of connections: 2
```

Once again what's reducing the number of connection is probably the 60 seconds `IdleTimeout`.
What am I missing?

Reply:
And also, I'm a bit concerned with the way the caching is being done in https://github.com/kubevirt/kubevirt/commit/3104bb01af4cc21fae11b3929a3d777352aa0a05. Let's say the virt-handler pod updates / restarts - I guess that we should re-establish the connection then. How can we know that the cached connection is still alive?

Reply:
Most likely that is because you did not use locks, but I  made a different suggestion which I would recommend to try out.